packages/core/src/components/card/card.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import * as classNames from "classnames";
+import * as PureRender from "pure-render-decorator";
+import * as React from "react";
+import * as Classes from "../../common/classes";
+import { IProps } from "../../common/props";
+import { safeInvoke } from "../../common/utils";
+
+export type ElevationSize = 0 | 1 | 2 | 3 | 4;
+
+export interface ICardProps extends IProps {
+    /**
+     * Elevation size of drop shadow that simulates card's height in the UI.
+     * @default 0
+     */
+    elevation?: ElevationSize;
+
+    /**
+     * Whether card on hover should change mouse to a pointer
+     * and the elevation shadow on the card increases by two levels.
+     * @default false
+     */
+    interactive?: boolean;
+
+    /**
+     * If set to `true`, all of the card's children will be converted into [skeletons](#core/components/progress/skeleton).
+     * @default false
+     */
+    loading?: boolean;
```

Comment:
👎 I'm incredibly wary of the implementation of this prop... Always found it brittle to modify `children` in any way. 

I'm inclined to nix this feature entirely and leave it up to consumers to do it themselves. Downside of this feature is that we could easily break their `children`, upside of not shipping it is that the consumer gets to ensure the correct behavior every time.

Reply:
yeah, I agree that modifying children smells hacky. I will not mind to remove this feature, but first let's think about other ways we can do it.

1. Do you think better test coverage for multiple scenarios of nested elements can make brittle code more safe? If we write tests for most of use-cases, then at least brittle code is under our control.

2. Maybe there is another way to "replace" components with skeletons? I was thinking about doing kind of dump of card's inner structure and modify it instead, but that's another messy solution. We can also randomly generate skeletons, but that sucks from the UX perspective. Have you guys have other ideas?

3. Maybe loading can be done without using skeletons at all? In my project I've made this kind of loading for card:
![spinner-loader](https://user-images.githubusercontent.com/2925620/31745271-5838426e-b462-11e7-955d-258d44cf05cb.gif)

Reply:
That loading card transition is very cool! 

However I think that loading is simply out of scope for this `Card` component. It should be a lightweight wrapper around the CSS API and nothing more. 

A more general Skeleton solution should should be just that: more general. Definitely not baked into this one component, but available to be composed with any other BP component(s). This was the logic behind providing a single `.pt-skeleton` class instead of a full React component: slap the class on any element and _voila_ it's loading.

Reply:
I agree with you, let's keep the Card component simple. Consumers can always prepare two kinds of card's content, one with real data and one "skeletonized" and change them accordingly

Reply:
exactly! glad we're in agreement.